examples/miscellaneous/plot_pipeline_display.py

```diff
@@ -0,0 +1,203 @@
+from sklearn.preprocessing import StandardScaler, PolynomialFeatures
+from sklearn.linear_model import LogisticRegression
+from sklearn import set_config
+
+steps = [
+    ("polynomial", PolynomialFeatures()),
+    ("standard_scaler", StandardScaler()),
+    ("classifier", LogisticRegression()),
+]
+pipe = Pipeline(steps)
+
+# To visualize the diagram, change to display='diagram'
+set_config(display="diagram")
+pipe
+
+# %%
+# Displaying a `Pipeline` and Dimensionality Reduction and Classifier
+###############################################################################
+# This section constructs a pipeline and displays its text representation.
+# HTML Illustration of and `PCA` and `SVC`
+
+from sklearn.pipeline import Pipeline
+from sklearn.svm import SVC
+from sklearn.decomposition import PCA
+from sklearn import set_config
+
+steps = [("reduce_dim", PCA()), ("classifier", SVC())]
+pipe = Pipeline(steps)
+
+# %%
+# To visualize the diagram, change to `display='diagram'`
+set_config(display="diagram")
+pipe
+
+# %%
+# Displaying a Complex `Pipeline` Chaining a Column Transformer
+###############################################################################
+# HTML Illustration of `Pipeline` and `ColumnTransformer` and `LogisticRegression`
+
+from sklearn.compose import make_column_selector as selector
+from sklearn.preprocessing import OneHotEncoder, StandardScaler
+from sklearn.compose import ColumnTransformer
+from sklearn.linear_model import LogisticRegression
+from sklearn.pipeline import make_pipeline
+import pandas as pd
+from sklearn import set_config
+
+numerical_columns_selector = selector(dtype_exclude=object)
+categorical_columns_selector = selector(dtype_include=object)
+
+data = pd.DataFrame()
+
+numerical_columns = numerical_columns_selector(data)
+categorical_columns = categorical_columns_selector(data)
+
+categorical_preprocessor = OneHotEncoder(handle_unknown="ignore")
+numerical_preprocessor = StandardScaler()
+
+preprocessor = ColumnTransformer(
+    [
+        ("one-hot-encoder", categorical_preprocessor, categorical_columns),
+        ("standard-scaler", numerical_preprocessor, numerical_columns),
+    ]
+)
+
+pipe = make_pipeline(preprocessor, LogisticRegression(max_iter=500))
+
+# %%
+# To visualize the diagram, change to `display='diagram'`
+set_config(display="diagram")
+pipe
+
+# %%
+# Displaying a Grid Search over a Pipeline
+###############################################################################
+# HTML Illustration of `GridSearchCV` over a `Pipeline` with `RandomForest`
+
+from sklearn.compose import make_column_selector as selector
+from sklearn.model_selection import GridSearchCV
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.impute import SimpleImputer
+from sklearn.preprocessing import OneHotEncoder, StandardScaler
+from sklearn.compose import ColumnTransformer
+from sklearn.pipeline import Pipeline
+from sklearn.pipeline import make_pipeline
```

Comment:
Some of these imports are duplicated and are already imported above. I see the value of having a self-contained notebook cell for this specific gallery example, but in general we don't do that (apart in the release highights). I think it's fine to leave as is if we decide that the cells should be self-contained?

Reply:
Yes, that's the reason the imports are duplicated.  So each example could stand alone on its own.